File: support_tools/swigwin-3.0.12/Examples/test-suite/ocaml/throw_exception_runme.ml
(* Throw exception test *)

open Swig
open Throw_exception

let x = new_Foo C_void ;;
let _ =
  try
    (invoke x) "test_int" C_void 
  with (Failure "Exception(37): Thrown exception from C++ (int)\n") ->
  try 
    (invoke x) "test_msg" C_void
  with (Failure "Exception(0): Dead\n") ->
  try
    (invoke x) "test_cls" C_void 
  with (Failure "Exception(0): Thrown exception from C++ (unknown)\n") ->
  try
    (invoke x) "test_multi" (C_int 1)
  with (Failure "Exception(37): Thrown exception from C++ (int)\n") ->
  try
    (invoke x) "test_multi" (C_int 2)
  with (Failure "Exception(0): Dead\n") ->
  try
    (invoke x) "test_multi" (C_int 3)
  with (Failure "Exception(0): Thrown exception from C++ (unknown)\n") ->
    exit 0

let _ = exit 1

File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/throw_exception_runme.cs
using System;
using throw_exceptionNamespace;

public class runme
{
    static void Main() {
        Foo f = new Foo();
        try {
            f.test_int();
            throw new Exception("Integer exception should have been thrown");
        } catch (System.Exception) {
        }
        try {
            f.test_msg();
            throw new Exception("String exception should have been thrown");
        } catch (System.Exception) {
        }
        try {
            f.test_cls();
            throw new Exception("Class exception should have been thrown");
        } catch (System.Exception) {
        }
    }
}
